src/MoBi.Engine/Sbml/CompartmentImporter.cs

```diff
@@ -116,13 +118,19 @@ private void CreateMoleculeProperties(IContainer container)
       /// <summary>
       ///     Creates a Volume Parameter for a MoBi Container. 
       /// </summary>
-      private IEntity CreateVolumeParameter()
+      private IEntity CreateVolumeParameter(Compartment compartment)
       {
-         IFormula formula = ObjectBaseFactory.Create<ConstantFormula>().WithValue(1);
+         var volume = 1.0;
+         if (compartment.isSetVolume())
+            volume = compartment.getVolume();
+         else if (compartment.isSetSize())
+            volume = compartment.getSize();
+         var dimension = _dimensionFactory.Dimension(Constants.Dimension.VOLUME);
+         IFormula formula = _formulaFactory.ConstantFormula(volume, dimension);
 
          var volumeParameter = _objectBaseFactory.Create<IParameter>()
             .WithName(SBMLConstants.VOLUME)
-            .WithDimension(_dimensionFactory.Dimension(OSPSuite.Core.Domain.Constants.Dimension.VOLUME))
+            .WithDimension(dimension)
```

Comment:
I miss the unit definition for the volume parameter.
As far as I understand this should be the default unit of the "volume" unit definition?

```
<unitDefinition id="volume" name="volume" metaid="volume">
        <listOfUnits>
          <unit metaid="CDMT00051" kind="litre"/>
        </listOfUnits>
</unitDefinition>
...
<listOfCompartments>
      <compartment id="default" metaid="default" sboTerm="SBO:0000290" units="volume" size="1"/>
      <compartment id="c2" name="Environment" metaid="c2" sboTerm="SBO:0000290" units="volume" outside="default" size="1"/>
    </listOfCompartments>
```

Reply:
@Yuri05 is right. The unit should be set as unit for the parameter. Now, is the value 1 in litre? Then we need to make sure we always convert the value in baseunit. @abdelr Is this done somewhere?

Reply:
Definitely!! Nice catch. Thanks.